Cython/Compiler/ExprNodes.py

```diff
@@ -849,6 +845,53 @@ def free_subexpr_temps(self, code):
     def generate_function_definitions(self, env, code):
         pass
 
+    # ----Generation of small bits of reference counting --
+
+    #@property
+    #def refcounting_type(self):
+        # FIXME this might just be "ctype"?
+    #    type = self.type
+    #    if self.is_temp and self.type.is_pyobject:
+            # when a temp object is created it's always just
+            # created as a py_object_type and not a
+            # derived type
+    #        type = PyrexTypes.py_object_type
+    #    return type
+
+    def generate_decref_set(self, code, rhs):
+        code.put_decref_set(self.result(), self.ctype(), rhs)
+
+    def generate_xdecref_set(self, code, rhs):
+        code.put_xdecref_set(self.result(), self.ctype(), rhs)
+
+    def generate_guarded_decref_set(self, code, rhs):
+        # TODO don't like the name
```

Comment:
What about making it a keyword argument to `generate_decref_set()`? E.g. `handle_null=True`?